### ISSUE TITLE
[RFC] Support --datarootdir

### ISSUE THREAD

**Issue by orbea** [mesonbuild/meson#7616]
**Describe the bug**

Meson does not support `--datarootdir`.

Meson only support `--datadir` and not `--datarootdir`, while often these are redundant, there are relatively common cases where this is important.

For example the emulator `snes9x` uses `--datadir` to install the desktop file, icons and locales, on Slackware these must be installed to `/usr/share/applications`, `/usr/share/icons` and `/usr/share/locale` respectively. However they also install some data in the `/usr/share/snes9x` directory which would be preferable to install to `/usr/share/games/snes9x` instead.

Before when snes9x used autotools I could leave `--datarootdir` to the default of `/usr/share` and set `--datadir=/usr/share/games` to get the intended behavior. Now snes9x has worked around this by adding their own `appdatadir` option where I set `-Dappdatadir=share/games/snes9x`. Suffice to say this is less than ideal and it should not have to be added to every applicable project one by one.

https://github.com/snes9xgit/snes9x

**Expected behavior**

Meson should support `--datarootdir` and not force projects to reinvent the wheel every time its applicable.

**system parameters**

OS: `Slackware64-current`
python: `3.8.1`
meson: `0.55.1`

**Comment by rubyFeedback** [comment]
I think this would be useful to have (and documented so people can find it) - could help some GNU configure
based projects to move towards meson too, so +1 to the suggestion.

I should however had say that using --datarootdir is quite rare. I rarely need it, but again, not reasoning against
the suggestion - the more modular meson can be, the better IMO.

**Comment by orbea** [comment]
I agree I more often do not need it than need it, but when I need do need `--datarootdir` its indispensable.
